components/sup/src/manager.rs

```diff
@@ -649,6 +649,8 @@ impl Manager {
             }
             Err(err) => {
                 outputln!("Unable to start {}, {}", ident, err);
+                // Remove the spec file so it does not look like this service is loaded.
+                self.try_remove_spec_file(&ident);
```

Comment:
> What happens if we get here and self.try_remove_spec_file fails?

We would end up in the wierd state this PR fixes. The state is that `hab svc status` would say that the service is not loaded, but when you go to load the service it would say it was already loaded. This is because the check that a service is loaded just checks that the spec path exists not that it is valid. Note, this would only happen if the `self.try_remove_spec_file` failed and the file still exists. If it failed because it somehow got removed we would be fine.

> Would we possibly be better off removing any existing spec file up front and not trying to start the service if that fails?

I do not think this would work. We use changes to the spec file to trigger behavior. If the spec file is removed, we interpret that as a signal [to shutdown the service](https://github.com/habitat-sh/habitat/blob/cfebef5e7e816a9720274830298e254c69eac3f3/components/sup/src/manager.rs#L1560). I believe this is the same as unloading a service. If we did this, we would be unloading every service when we tried to start them.

Reply:
Instead of completely removing a bad spec file, I wonder if it might be useful to move it to another location for subsequent inspection by an operator? Maybe not... just a thought :thinking: 

If a spec file is rejected, are all the reasons why conveyed to the user (via the CLI and/or appropriate logging?)

Reply:
> Instead of completely removing a bad spec file, I wonder if it might be useful to move it to another location

Where would be a reasonable place to move it? It seems like this might fall under YAGNI. This should happen very rarely, but when it does it would be very useful to know why. I dont have a strong preference either way?

> If a spec file is rejected, are all the reasons why conveyed to the user

Yes, if `service_load` fails they will be immediately notified at the cli and the service will not be loaded. If due to updates (or any other reason), `Service::new` fails later on in `add_service_mlw` the reason for the failure is logged and the service is automatically unloaded.

Reply:
Yeah, probably YAGNI... if we need it, it's easy enough to add.